common/src/tap/checks/sender_balance_check.rs

```diff
@@ -38,22 +38,25 @@ impl Check for SenderBalanceCheck {
             .recover_signer(&self.domain_separator)
             .inspect_err(|e| {
                 error!("Failed to recover receipt signer: {}", e);
-            })?;
+            })
+            .map_err(|e| CheckError::Failed(e.into()))?;
 
         // We bail if the receipt signer does not have a corresponding sender in the escrow
         // accounts.
-        let receipt_sender = escrow_accounts_snapshot.get_sender_for_signer(&receipt_signer)?;
+        let receipt_sender = escrow_accounts_snapshot
+            .get_sender_for_signer(&receipt_signer)
+            .map_err(|e| CheckError::Failed(e.into()))?;
 
         // Check that the sender has a non-zero balance -- more advanced accounting is done in
         // `tap-agent`.
         if !escrow_accounts_snapshot
             .get_balance_for_sender(&receipt_sender)
             .map_or(false, |balance| balance > U256::ZERO)
         {
-            return Err(anyhow!(
+            return Err(CheckError::Failed(anyhow!(
```

Comment:
This can be retryable. Throwing away all the receipts until the sender tops off their balance is not helpful. Better wait for it. We'll keep track of the unaggregated fees either way.

Reply:
The only problem is that the check is on the "common" side, not the tap-agent side. We don't accept it if it fails either way, should we?

Reply:
Also, this specifc test is just to check if we have a balance greater than zero. Most of the time, this will be someone who never deposited. This could open a precedence of people just creating senders without depositing anything and using the max amount willing to lose.

Reply:
Ah yeah you're right, this is the wrong balance check for this